gs_csr.json

```diff
@@ -0,0 +1,42 @@
+[
+  [
+    {
+      "Name" : "CSR_Gauss-siedel",
+      "Input Type" : "IEgenLIB Relation",
+      "Relation" : "{ [i,j] -> [ip,jp] : i<ip && i=colidx(jp) && 0 <= i && i < m && 0 <= ip && ip < m && rowptr(i) <= j && j < rowptr(i+1) && rowptr(ip) <= jp && jp < rowptr(ip+1) }",
+      "Expected" : "{ [i] -> [ip,jp] : i=colidx(jp) && colidx(jp) < ip && ip < m-1 && rowptr(i) < rowptr(i+1) && rowptr(ip) <= jp && jp < rowptr(ip+1) && colidx(jp) >= 0 && rowptr(i) >= 0 && rowptr(ip) >= 0 && rowptr(i+1) < nnz && rowptr(ip+1) < nnz}",
+      "UFS" :
+      [
+        { 
+           "Name" : "colidx",
+           "Domain" : "{[i]:0<=i &&i<nnz}",
+           "Range" : "{[j]:0<=j &&j<m}",
+           "Bijective" : "false",
+           "Monotonicity" : "iegenlib::Monotonic_NONE"
+        },
+        { 
+           "Name" : "rowptr",
+           "Domain" : "{[j]:0<=j &&j<m}",
+           "Range" : "{[i]:0<=i &&i<nnz}",
+           "Bijective" : "false",
+           "Monotonicity" : "iegenlib::Monotonic_Increasing"
+        }
+      ],
+      "User Defined" :
+      [
+
+      ],
+      "Do Not Project" :
+      [
+        "0", "2"
```

Comment:
I changed these acording to comments.

On Thu, Jun 23, 2016 at 8:48 PM, Michelle Strout <notifications@github.com>
wrote:

> In gs_csr.json
> <https://github.com/CompOpt4Apps/IEGenLib/pull/38#discussion_r68265557>:
>
> > +        },
> > +        {
> > +           "Name" : "rowptr",
> > +           "Domain" : "{[j]:0<=j &&j<m}",
> > +           "Range" : "{[i]:0<=i &&i<nnz}",
> > +           "Bijective" : "false",
> > +           "Monotonicity" : "iegenlib::Monotonic_Increasing"
> > +        }
> > +      ],
> > +      "User Defined" :
> > +      [
> > +
> > +      ],
> > +      "Do Not Project" :
> > +      [
> > +        "0", "2"
>
> Let's change the "Do Not Project" to "i" and "ip". Also call it, "Do Not
> Project Out" or "Do Not Eliminate".
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CompOpt4Apps/IEGenLib/pull/38/files/873a894609e6a1eb8305ba539144baac69001790#r68265557>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ANGVl70IZJQClBf_At8Xgxk8rgqvkhopks5qOrHAgaJpZM4I89jG>
> .
>